core/src/storages/secdist/component.cpp

```diff
@@ -2,64 +2,90 @@
 
 #include <userver/components/component.hpp>
 #include <userver/logging/log.hpp>
+#include <userver/storages/secdist/default_provider.hpp>
 #include <userver/storages/secdist/exceptions.hpp>
-#include <userver/storages/secdist/provider.hpp>
 #include <userver/utils/string_to_duration.hpp>
 #include <userver/yaml_config/merge_schemas.hpp>
 
 USERVER_NAMESPACE_BEGIN
-
+/// [Sample secdist - default secdist]
 namespace components {
-
 namespace {
 
+storages::secdist::SecdistFormat FormatFromString(std::string_view str) {
+  if (str.empty() || str == "json") {
+    return storages::secdist::SecdistFormat::kJson;
+  } else if (str == "yaml") {
+    return storages::secdist::SecdistFormat::kYaml;
+  }
+
+  UINVARIANT(
+      false,
+      fmt::format("Unknown secdist format '{}' (must be one of 'json', 'yaml')",
+                  str));
+}
+
 storages::secdist::SecdistConfig::Settings ParseSettings(
     const ComponentConfig& config, const ComponentContext& context) {
-  storages::secdist::SecdistConfig::Settings settings;
+  using Provider = storages::secdist::DefaultProvider;
+  Provider::Settings provider_settings;
+  provider_settings.config_path = config["config"].As<std::string>({});
+  provider_settings.format =
+      FormatFromString(config["format"].As<std::string>({}));
+  provider_settings.missing_ok = config["missing-ok"].As<bool>(false);
+  provider_settings.environment_secrets_key =
+      config["environment-secrets-key"].As<std::optional<std::string>>();
+  auto blocking_task_processor_name =
+      config["blocking-task-processor"].As<std::optional<std::string>>();
+  provider_settings.blocking_task_processor =
+      blocking_task_processor_name
+          ? &context.GetTaskProcessor(*blocking_task_processor_name)
+          : nullptr;
 
-  const auto provider_name =
-      config["provider"].As<std::string>("default-secdist-provider");
```

Comment:
Wow! You've done much more than we were expecting!

Unfortunately, in our internal codebase we already have a lot of custom providers. It is almost impossible for us, to make a migration in one go - hundreds of services should be adjusted and changes simultaneously.

Could you please split this PR into multiple?

First PR should preserve the `provider` config parameter and the `SecdistProvider` hierarchy. The `DefaultSecdistProvider` component should remain in code but not used in userver static configs. `provider` option should become empty by default, and in that case the whole functionality is implemented by the Secdist itself. That way we could merge the PR without affecting the services, remove the `default-secdist-provider` usage in hundreds of services.

After that, the `DefaultSecdistProvider` could be removed by a separate PR. And after that, we can take a second look on new hierarchy of providers (however, I think that it is fine to leave the `provider` to customize the Secdist in complicated cases).